src/units/abilities.cpp

```diff
@@ -2134,7 +2134,7 @@ effect::effect(const active_ability_list& list, int def, const const_attack_ptr&
 	for(const active_ability& i : list) {
 		double priority = i.ability().priority();
 		if(base_list[priority].empty()) {
-			base_list[priority] = list.loc();
+			base_list[priority] = active_ability_list(list.loc());
 		}
 		base_list[priority].emplace_back(i);
 	}
```

Comment:
This whole loop could be tidied up. How about:
```suggestion
	for(const active_ability& i : list) {
		auto [iter, _] = base_list.try_emplace(i.ability().priority(), list.loc());
		active_ability_list& priority_bucket = iter->second;
		priority_bucket.emplace_back(i);
	}
```

Reply:
Soliton's version looks more readable to me.

Reply:
I'm not really interested in adding random further commits to this PR. That can be done in a different PR. I agree that the tidied up version does not look particularly readable though the current code is not great either.